Yank/storage/iodrivers.py

```diff
@@ -0,0 +1,1753 @@
+        # Check writeable
+        if self._output_mode != 'w':
+            raise TypeError("{} at {} was saved as appendable data! Cannot overwrite, must use append()".format(
+                self.dtype_string, self._target)
+            )
+        # Save data
+        self._bound_target[:] = self._encoder(data)
+        return
+
+    def append(self, data):
+        # Check type
+        if type(data) is not self._dtype:
+            raise TypeError("Invalid data type on variable {}.".format(self._target))
+        # Bind
+        if self._bound_target is None:
+            self._bind_append(data)
+        # Check writeable
+        if self._output_mode != 'a':
+            raise TypeError("{} at {} was saved as appendable data! Cannot overwrite, must use append()".format(
+                self.dtype_string, self._target)
+            )
+        # Determine current current length and therefore the last index
+        length = self._bound_target.shape[0]
+        # Save data
+        self._bound_target[length, :] = self._encoder(data)
+
+    @property
+    def _storage_type(self):
+        return 'variables'
+
+
+class NCInt(NCScalar):
+    """
+    NetCDF handler for Integers
+    """
+
+    @property
+    def _encoder(self):
+        return nc_int_encoder
+
+    @property
+    def _decoder(self):
+        return nc_int_decoder
+
+    @property
+    def _dtype(self):
+        return int
+
+    @staticmethod
+    def _dtype_string():
+        return "int"
+
+
+class NCFloat(NCScalar):
+    """
+    NetCDF handler for Floats
+    """
+
+    @property
+    def _encoder(self):
+        return nc_float_encoder
+
+    @property
+    def _decoder(self):
+        return nc_float_decoder
+
+    @property
+    def _dtype(self):
+        return float
+
+    @staticmethod
+    def _dtype_string():
+        return "float"
+
+
+class NCString(NCScalar):
+    """
+    NetCDF handler for String
+    """
+
+    @property
+    def _encoder(self):
+        return nc_string_encoder
+
+    @property
+    def _decoder(self):
+        return nc_string_decoder
+
+    @property
+    def _dtype(self):
+        return str
+
+    @staticmethod
+    def _dtype_string():
+        return "str"
+
+
+# Array
+
+class NCArray(NCVariableTypeHandler):
+    """
+    NetCDF handler for numpy arrays
+    """
+    @property
+    def _dtype(self):
+        return np.ndarray
+
+    @staticmethod
+    def _dtype_string():
+        return "numpy.ndarray"
+
+    def _bind_read(self):
+        try:
+            self._attempt_storage_read()
+            # Handle variable size objects
+            # This line will not happen unless target is real, so None should not be returned by _check_write_append
+            self._check_write_append()
+            if self._output_mode is 'a':
+                self._save_shape = self._bound_target.shape[1:]
+            else:
+                self._save_shape = self._bound_target.shape
+        except KeyError as e:
+            raise e
```

Comment:
Isn't this equivalent to not have a `try..except` clause? Same in `NCScalar` and `NCIterable`.